vm/devices/vmbus/vmbus_server/src/channels/saved_state.rs

```diff
@@ -972,3 +996,107 @@ enum GpadlState {
     #[mesh(4)]
     TearingDown,
 }
+
+#[derive(Debug, Clone, Protobuf, PartialEq, Eq)]
+#[mesh(package = "vmbus.server.channels")]
+struct ConnectionTarget {
+    #[mesh(1)]
+    pub vp: u32,
+    #[mesh(2)]
+    pub sint: u8,
+}
+
+impl ConnectionTarget {
+    fn save(value: &super::ConnectionTarget) -> Self {
+        Self {
+            vp: value.vp,
+            sint: value.sint,
+        }
+    }
+
+    fn restore(self) -> super::ConnectionTarget {
+        super::ConnectionTarget {
+            vp: self.vp,
+            sint: self.sint,
+        }
+    }
+}
+
+#[derive(Debug, Clone, Protobuf, PartialEq, Eq)]
+#[mesh(package = "vmbus.server.channels")]
+enum MessageTarget {
+    #[mesh(1)]
+    Default,
+    #[mesh(2)]
+    ReservedChannel(u32),
```

Comment:
I think you'll find that this just won't work, and we should only save things targeted at `Default`.